provider/azure/azure.go

```diff
@@ -160,9 +161,9 @@ func (p *AzureProvider) ApplyChanges(ctx context.Context, changes *plan.Changes)
 	}
 
 	deleted, updated := p.mapChanges(zones, changes)
-	p.deleteRecords(ctx, deleted)
-	p.updateRecords(ctx, updated)
-	return nil
+	err1 := p.deleteRecords(ctx, deleted)
+	err2 := p.updateRecords(ctx, updated)
+	return errors.Join(err1, err2)
```

Comment:
Are we generally fine with the change of semantic of how we deal the error? Bubbling this back up has obvious consequences.

Reply:
Using errors.join makes it easier to see where an error occurred. 
WDYT? @mloiseleur @johngmyers @szuecs

Reply:
This doesn’t answer my question. This new error handling will definitely change the way metrics are reported, but also the semantics on what happens when we reach an error. I am not so sure this won’t be a surprise for some users, although we can say that erroring “the hard way” is most likely always the right thing.

Reply:
> This doesn’t answer my question. This new error handling will definitely change the way metrics are reported, but also the semantics on what happens when we reach an error. I am not so sure this won’t be a surprise for some users, although we can say that erroring “the hard way” is most likely always the right thing.

Thanks @Raffo  WDYT? 
like:
```
if err := p.deleteRecords(ctx, deleted); err != nil {
    return err
}

if err := p.updateRecords(ctx, updated); err != nil {
    return err
}

return nil
```

Reply:
This is a different change in a different direction. You won't execute `updateRecords` in case `deleteRecords` fail and I am not sure this is the intended change. I think we should not proceed with this change as is unless you can provide a clear picture of what the error handling from the provider should be and what the consequences could be for the users of the provider.